crates/ruff_workspace/src/resolver.rs

```diff
@@ -92,40 +92,32 @@ impl Relativity {
 
 #[derive(Default)]
 pub struct Resolver {
-    settings: BTreeMap<PathBuf, AllSettings>,
+    settings: BTreeMap<PathBuf, Settings>,
 }
 
 impl Resolver {
     /// Add a resolved [`Settings`] under a given [`PathBuf`] scope.
-    fn add(&mut self, path: PathBuf, settings: AllSettings) {
+    fn add(&mut self, path: PathBuf, settings: Settings) {
         self.settings.insert(path, settings);
     }
 
-    /// Return the appropriate [`AllSettings`] for a given [`Path`].
-    pub fn resolve_all<'a>(
+    /// Return the appropriate [`Settings`] for a given [`Path`].
+    pub fn resolve<'a>(
         &'a self,
         path: &Path,
         pyproject_config: &'a PyprojectConfig,
-    ) -> &'a AllSettings {
+    ) -> &'a Settings {
         match pyproject_config.strategy {
-            PyprojectDiscoveryStrategy::Fixed => &pyproject_config.settings,
+            PyprojectDiscoveryStrategy::Fixed => &pyproject_config.settings.lib,
             PyprojectDiscoveryStrategy::Hierarchical => self
                 .settings
                 .iter()
                 .rev()
                 .find_map(|(root, settings)| path.starts_with(root).then_some(settings))
-                .unwrap_or(&pyproject_config.settings),
+                .unwrap_or(&pyproject_config.settings.lib),
```

Comment:
@charliermarsh what's the resolution logic here? It feels kind of slow because it iterates over all settings without making use of the `BTreeMap`. Do you think we could optimize it by using `range` to prefilter the potential settings?

Reply:
I think the intent here is just to iterate from longest path to shortest. I'm not sure if that answers your question...